pkg/config/release.go

```diff
@@ -163,6 +183,278 @@ func GetChangedTemplates(path, baseRev string) ([]ConfigMapSource, error) {
 	return ret, nil
 }
 
+func loadRegistryStep(filename string, fullRegistry registry) (RegistryStep, error) {
+	if strings.HasSuffix(filename, "-commands.sh") {
+		filename = strings.TrimSuffix(filename, "-commands.sh")
+		filename = strings.Join([]string{filename, "-ref.yaml"}, "")
+	}
+	if strings.HasSuffix(filename, "-ref.yaml") {
+		name := strings.TrimSuffix(filename, "-ref.yaml")
+		ref, ok := fullRegistry.refs[name]
+		if !ok {
+			return RegistryStep{}, fmt.Errorf("could not find reference: %s", name)
+		}
+		change := RegistryStep{
+			Reference: &api.RegistryReference{
+				LiteralTestStep: ref,
+			},
+		}
+		return change, nil
+	}
+	if strings.HasSuffix(filename, "-chain.yaml") {
+		name := strings.TrimSuffix(filename, "-chain.yaml")
+		chain, ok := fullRegistry.chains[name]
+		if !ok {
+			return RegistryStep{}, fmt.Errorf("could not find chain: %s", name)
+		}
+		change := RegistryStep{
+			Chain: &api.RegistryChain{
+				As:    name,
+				Steps: chain,
+			},
+		}
+		return change, nil
+	}
+	if strings.HasSuffix(filename, "-workflow.yaml") {
+		name := strings.TrimSuffix(filename, "-workflow.yaml")
+		workflow, ok := fullRegistry.workflows[name]
+		if !ok {
+			return RegistryStep{}, fmt.Errorf("could not find workflow: %s", name)
+		}
+		change := RegistryStep{
+			Workflow: &api.RegistryWorkflow{
+				As:    name,
+				Steps: workflow,
+			},
+		}
+		return change, nil
+	}
+	return RegistryStep{}, fmt.Errorf("invalid step filename: %s", filename)
+}
+
+// GetChangedRegistrySteps identifies all registry components (refs, chains, and workflows) that changed.
+// After all changes have been identified, the list gets "trimmed": all changed refs that are included in
+// changed chains are removed from the list since the refs will be tested when the chain gets tested. This
+// is also done for changed chains and refs included in changed workflows. This helps reduce the number of
+// rehearsals that are performed.
+func GetChangedRegistrySteps(path, baseRev string, refs map[string]api.LiteralTestStep, chains map[string][]api.TestStep, workflows map[string]api.MultiStageTestConfiguration) ([]RegistryStep, error) {
+	revChanges, err := getRevChanges(path, RegistryPath, baseRev, true)
+	if err != nil {
+		return nil, err
+	}
+	var registryChanges []RegistryStep
```

Comment:
this seems misplaced? or should it have had a load line? Why is there an error check afterword?